cmd/kubeadm/app/kubeadm.go

```diff
@@ -22,15 +22,14 @@ import (
 
 	"github.com/spf13/pflag"
 
-	"k8s.io/klog/v2"
-
 	cliflag "k8s.io/component-base/cli/flag"
 	"k8s.io/kubernetes/cmd/kubeadm/app/cmd"
+	// package k8s.io/component-base/log is responsible for doing klog.InitFlags()
```

Comment:
This is not true, currently code for registering klog flags is located in 
https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/component-base/cli/globalflag/globalflags.go

As Kubeadm doesn't use either `k8s.io/component-base/cli` nor uses `k8s.io/component-base/logs` let's leave their klog invocation as it is.

Reply:
Indirectly import the package  `imports k8s.io/component-base/logs`
```
package k8s.io/kubernetes/cmd/kubeadm/app
	imports k8s.io/kubernetes/cmd/kubeadm/app/cmd
	imports k8s.io/kubernetes/cmd/kubeadm/app/apis/kubeadm/validation
	imports k8s.io/kubernetes/cmd/kubeadm/app/componentconfigs
	imports k8s.io/kube-proxy/config/v1alpha1
	imports k8s.io/component-base/config/v1alpha1
	imports k8s.io/component-base/config/options
	imports k8s.io/component-base/logs
```
If the code does not change, an error is reported
https://github.com/kubernetes/kubernetes/pull/99693#discussion_r592033489

Is that what you're saying ?
```
	local := flag.NewFlagSet(os.Args[0], flag.ExitOnError)
	klog.InitFlags(local)
```

Reply:
> 	imports k8s.io/kube-proxy/config/v1alpha1
>	imports k8s.io/component-base/config/v1alpha1
>	imports k8s.io/component-base/config/options
>	imports k8s.io/component-base/logs

this is a problematic import that is the result of this PR and we need to resolve it.

this means that all consumers of the public kube-proxy API type or the `config/v1alpha1` type directly, will get their default CommandLine mutated...

we have to either make `component-base/config/v1alpha1` not use the `logs` package, or make the `logs` package not call `InitFlags()` in `init()`.

/hold

Reply:
Thank you very much for your advice. I'm more inclined to delete this init method. Because it's really difficult to find out which package to import log package and call init function...

Reply:
i think it's worth investigating when it was added (probably near the klog v2 migration?) and see what can break if it's removed.

Reply:
Yes, I need to look at this part of the code to see why it was added at the time.

Reply:
> i think it's worth investigating when it was added (probably near the klog v2 migration?)
https://github.com/kubernetes/kubernetes/pull/70889/

This PR #70889 adds staging/src/k8s.io/component-base/logs/logs.go package klog.initFlags (Flag.mandline) code.
```
klog.initFlags (Flag.mandline)
```

And that's when cmd/kubeadm/kubeadm.go adds klog.InitFlag (nil) https://github.com/dims/kubernetes/blob/e558e291d1a41728da23f517e51b71038e3ba93e/cmd/kubeadm/kubeadm.go#L28

> see what can break if it's removed.

I tested what would happen if I removed the staging/src/k8s.io/component-base/logs/logs.go package init function, and there were a lot of errors. You can see that there are a lot of places that depend on the log package's klog.InitFlags(flag.CommandLine).
There are the tests logs:
https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/102326/pull-kubernetes-unit/1397473547823091712
https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/102326/pull-kubernetes-integration/1397482863271612416
https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/102326/pull-kubernetes-node-e2e-containerd/1397483544204283904/
https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/102326/pull-kubernetes-e2e-kind/1397483544095232000

Removing this method seems like a big job.